docs/Command Reference.md

```diff
@@ -0,0 +1,90 @@
+* `hokusai dev logs` - Print logs from the development stack.
+* `hokusai dev shell` - Attach a shell session to the stack's primary project container.
+* `hokusai dev clean` - Stop and remove all containers in the stack.
+
+### Using Hokusai for testing / CI
+
+* `hokusai test` - Start the testing stack defined `hokusai/test.yml` and exit with the return code of the test command.
+
+### Working with Images
+
+* `hokusai push` - Build and push an image to the AWS ECR project repo.
+* `hokusai images` - Print image builds and tags in the AWS ECR project repo.
+
+### Working with Kubernetes
+Hokusai uses `kubectl` to connect to Kubernetes. Hokusai `configure` provides basic setup for installing and configuring kubectl:
+```bash
+hokusai configure --help
+```
+Recommended approach is to upload your `kubectl` config to S3 and use following command to install it:
+```bash
+hokusai configure --kubectl-version <kubectl version> --s3-bucket <bucket name> --s3-key <file key>
+```
+
+When running `hokusai setup` `staging.yml` and `production.yml` are created in the `./hokusai` project directory. These files define what Hokusai refers to as "stacks", and Hokusai is opinionated about a workflow between a staging and production Kubernetes context.  Hokusai commands such as `stack`, `env`, `deploy` and `logs` require invocation with either the `--staging` or `--production` flag, which references the respective stack YAML file and interacts with the respective Kubernetes context.
+
+### Working with Environment Variables
+
+* `hokusai env create` - Create the Kubernetes configmap object `{project_name}-environment`
+* `hokusai env get` - Print envrionment variables stored on the Kubernetes server
+* `hokusai env set` - Set envrionment variables on the Kubernetes server. Environment variables are stored for the project as key-value pairs in the Kubernetes configmap object `{project_name}-environment`
+* `hokusai env unset` - Remove envrionment variables stored on the Kubernetes server
+* `hokusai env delete` - Delete the Kubernetes configmap object `{project_name}-environment`
+
+Note: Environment variables will be automatically injected into containers created by the `hokusai run` command but must be explicity referenced in the stack container yaml definition using `envFrom`.
+
+### Working with Stacks
+
+* `hokusai stack create` - Create a stack.
+* `hokusai stack update` - Update a stack.
+* `hokusai stack delete` - Delete a stack.
+* `hokusai stack status` - Print the stack status.
+
+### Working with Deployments
+
+* `hokusai deploy` - Update the Kubernetes deployment to a given image tag.
+* `hokusai promote` - Update the Kubernetes deployment on production to match the deployment running on staging.
+* `hokusai refresh` - Refresh the project's deployment(s).
+* `hokusai history` - Print the project's deployment(s) history.
+* `hokusai diff` - Print a git diff between the tags deployed on production vs staging
```

Comment:
In general I'd add a little more detail about these. For `promote`, you could clarify that the precise docker image from staging will be deployed. I'm not sure what `refresh` really does. And you could mention specifically what `history` prints (image tags? git SHAs? past deploy times?).